### PR TITLE
[DEV-5297] COVID Page ES-backed recipient location map

### PR DESCRIPTION
**Pull request by kwhickey**
**Description:**
Implementing `disaster/spending_by_geography` endpoint to shade the map by obligation/outlay/loan spending by recipient location.

**Technical details:**
This version of the implementation ports over the `search/spending_by_geography` approach from Advanced Search, and uses Elasticsearch to serve up the data, but targeting the awards elasticsearch index.

- WIP: 
    - [x] Ported over / stubbed out a good deal of the API view impl
    - [x] Added agg keys to `award_delta_view.sql`
    - [x] Some TODOs to reconcile
    - [x] agg key fields need to be added to the award ES index
    - [x] awards ES index needs to be reindexed to get new data in new fields
    - [x] CI: tests + formatting

**Requirements for PR merge:**

1. [x] Unit & integration tests updated
2. [x] API documentation updated
3. [ ] Necessary PR reviewers:
    - [x] Backend
    - [ ] Frontend <OPTIONAL>
    - [ ] Operations <OPTIONAL>
    - [ ] Domain Expert <OPTIONAL>
4. [x] Matview impact assessment completed
5. [x] Frontend impact assessment completed
6. [x] Data validation completed
7. [ ] Appropriate Operations ticket(s) created
8. [X] Jira Ticket [DEV-5297](https://federal-spending-transparency.atlassian.net/browse/DEV-5297):
    - [X] Link to this Pull-Request
    - [X] Performance evaluation of affected (API | Script | Download)
    - [X] Before / After data comparison

**Area for explaining above N/A when needed:**
```
```

### CODE REVIEW

File: usaspending_api/disaster/v2/views/spending_by_geography.py [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+    COUNTY = "county"
+    DISTRICT = "district"
+    STATE = "state"
+
+
+class SpendingByGeographyViewSet(DisasterBase):
+    """Spending by Recipient Location"""
+
+    endpoint_doc = "usaspending_api/api_contracts/contracts/v2/disaster/spending_by_geography.md"
+    # TODO: Are contracts under https://github.com/fedspendingtransparency/usaspending-api/tree/dev/usaspending_api/api_contracts/contracts/v2/disaster/recipient/location needed anymore??
```
Comment (tony-sappe) [inline usaspending_api/disaster/v2/views/spending_by_geography.py:32]: Those API Contracts are for the endpoints to power the table-view behind the map (post-MVP)

File: usaspending_api/disaster/v2/views/spending_by_geography.py [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+        self.geo_layer_filters = json_request.get("geo_layer_filters")
+        self.loc_field_name = location_dict[self.geo_layer.value]
+        self.loc_lookup = f"{self.scope_field_name}_{self.loc_field_name}"
+
+        # Set which field will be the aggregation amount
+        if self.filters["spending_type"] == "obligation":
+            self.metric_field = "total_covid_obligation"
+        elif self.filters["spending_type"] == "outlay":
+            self.metric_field = "total_covid_outlay"
+        elif self.filters["spending_type"] == "loan":
```
Comment (tony-sappe) [inline usaspending_api/disaster/v2/views/spending_by_geography.py:115]:
```suggestion
        elif self.filters["spending_type"] == "face_value_of_loan":
```

File: usaspending_api/settings.py [hunk 3]
```diff
@@ -41,7 +41,8 @@
 SECRET_KEY = get_random_string()
 
 # SECURITY WARNING: don't run with debug turned on in production!
-DEBUG = False
+# Defaults to False, unless DJANGO_DEBUG env var is set to a truthy value
+DEBUG = os.environ.get("DJANGO_DEBUG", "").lower() in ["true", "1", "yes"]
```
Comment (tony-sappe) [inline usaspending_api/settings.py:45]: @kwhickey need to verify there isn't a failsafe check in the deploy scripts which will fail or overwrite this

Comment (kwhickey) [reply usaspending_api/settings.py:45]: @tony-sappe It does a regex match for anything after `=`, so we're good here (always want it to be `False` for deployed envs (non local envs)).

https://github.com/fedspendingtransparency/data-act-build-tools/blob/master/usaspending-deploy/usaspending-api-launch.yml#L211-L216

File: .env [hunk 4]
```diff
@@ -27,3 +28,6 @@ BROKER_USER=root
 BROKER_PASSWORD=password
 BROKER_HOST=broker-db
 BROKER_PORT=5432
+
+# Default to having Django settings.py DEBUG=True for local docker envs
+DJANGO_DEBUG=True
```
Comment (tony-sappe) [inline .env:33]: [not critical] missing trailing newline

File: docker-compose.yaml [hunk 5]
```diff
@@ -63,7 +70,7 @@ services:
         - bootstrap.memory_lock=true
         - "ES_JAVA_OPTS=-Xms1536m -Xmx1536m"  # Ensure Docker is allocated plenty of memory, otherwise this will fail
       # Inject plugin install, then resume with orignial entrypoint command
-      command: /bin/sh -c "./bin/elasticsearch-plugin install mapper-murmur3 && /usr/local/bin/docker-entrypoint.sh"
+      command: /bin/sh -c "if [ ! -d /usr/share/elasticsearch/plugins/mapper-murmur3 ]; then ./bin/elasticsearch-plugin install mapper-murmur3; fi && /usr/local/bin/docker-entrypoint.sh"
```
Comment (tony-sappe) [inline docker-compose.yaml:73]: thanks @kwhickey !

**Pull request merged**